pg_wait_sampling.c

```diff
@@ -303,48 +219,45 @@ pgws_shmem_request(void)
 static void
 pgws_shmem_startup(void)
 {
-	bool		found;
-	Size		segsize = pgws_shmem_size();
-	void	   *pgws;
-	shm_toc	   *toc;
+	bool	found;
+	HASHCTL	info;
+
+	if (prev_shmem_startup_hook)
+		prev_shmem_startup_hook();
 
-	pgws = ShmemInitStruct("pg_wait_sampling", segsize, &found);
+	/* Create or attach to the shared memory state */
+	LWLockAcquire(AddinShmemInitLock, LW_EXCLUSIVE);
 
+	pgws_proc_queryids = ShmemInitStruct(
+			"pg_wait_sampling queryids",
+			sizeof(pgwsQueryId) * get_max_procs_count(),
+			&found);
+	MemSet(pgws_proc_queryids, 0, sizeof(pgwsQueryId) * get_max_procs_count());
```

Comment:
I think this MemSet should be in the `if (!found)` branch.

Reply:
Agreed. Thanks